### ISSUE TITLE
pm2 install from git

### ISSUE THREAD

**Issue by JulienPalard** [Unitech/pm2#2973]
**What's going wrong?**

`npm install git+https://github.com/JulienPalard/pm2-syslog.git` works but 
`pm2 install git+https://github.com/JulienPalard/pm2-syslog.git` don't. To use a patched version of a module while waiting the patch to be merged, this looks the only solution and would be a nice one but it throws:

```
[PM2][Module] Installing module git+https://github.com/JulienPalard/pm2-syslog.git
[PM2][Module] Calling [NPM] to install git+https://github.com/JulienPalard/pm2-syslog.git ...
+ pm2-syslog2@2.2.4
updated 1 package in 3.179s
[PM2][Module] Module downloaded
ENOENT: no such file or directory, open '/home/julien/.pm2/node_modules/package.json'
module.js:327
    throw err;
    ^

Error: Cannot find module '/home/julien/.pm2/node_modules/package.json'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at startModule (/home/julien/pm2-syslog2/node_modules/pm2/lib/API/Modules/Modularizer.js:33:22)
    at /home/julien/pm2-syslog2/node_modules/pm2/lib/API/Modules/Modularizer.js:164:9
    at /home/julien/pm2-syslog2/node_modules/pm2/lib/Configuration.js:64:14
    at FSReqWrap.oncomplete (fs.js:82:15)
```

**How could we reproduce this issue?**

Just type `pm2 install git+https://github.com/JulienPalard/pm2-syslog.git`

## Supporting information

```
PM2 version: `pm2 -v` 2.5.0
Node version: `node -v` v4.8.2
Windows? Mac? Linux? Debian GNU/Linux 9.0 (stretch)
```

**Comment by vmarchaud** [comment]
There is another problem that we can't really fix : 
- You are trying to install the package `pm2-syslog2` as registered in your package.json
- but you are trying to install it with the name `JulienPalard/pm2-syslog`

The module system allow to put configuration inside the module `package.json` so it can be picked up when PM2 launch it. To resolve it, it try to resolve the package.json location (using the link with which you are installing it), so since the package manager (either npm or yarn) put the module inside a folder with the name of the module, PM2 can't find the package.json.

I can put a fix in-place to don't crash when not finding the package.json, but i'll need add an error message if user understand what's going on.

In the mean-time, you can just use the development version : `npm install -g Unitech/pm2#development` and rename your github repository to make it works :)

**Comment by JulienPalard** [comment]
Thanks for the patch and the clear feedback. I only appended the "2" to allow me to push on npmjs as install with git+https was not working. Renaming will no longer be necessary with your patch so everything will work as expected. Thanks again.

**Comment by Unitech** [comment]
Published on PM2 2.6.0:

```bash
$ npm install pm2@latest -g
$ pm2 update
```
